src/libponyc/expr/reference.c

```diff
@@ -193,6 +193,44 @@ bool expr_field(pass_opt_t* opt, ast_t* ast)
   return true;
 }
 
+bool expr_contained_in(ast_t* ast, ast_t* expected_recover)
```

Comment:
I suggest making this static (since it's only used in this file) and/or making the name more specific to the fact that it only is suitable for checking if an expression is in a given recover block.

```suggestion
static bool expr_contained_in_recover(ast_t* ast, ast_t* expected_recover)
```